JetStreamCompose/jetstream/src/main/java/com/google/jetstream/presentation/screens/categories/CategoriesScreen.kt

```diff
@@ -49,18 +51,47 @@ import androidx.tv.material3.CardLayoutDefaults
 import androidx.tv.material3.MaterialTheme
 import androidx.tv.material3.StandardCardLayout
 import androidx.tv.material3.Text
-import com.google.jetstream.presentation.LocalMovieRepository
+import com.google.jetstream.data.entities.MovieCategoryList
 import com.google.jetstream.presentation.screens.dashboard.rememberChildPadding
 import com.google.jetstream.presentation.theme.JetStreamBorderWidth
 import com.google.jetstream.presentation.theme.JetStreamCardShape
 import com.google.jetstream.presentation.utils.GradientBg
 
-@OptIn(ExperimentalComposeUiApi::class)
 @Composable
 fun CategoriesScreen(
     gridColumns: Int = 4,
     onCategoryClick: (categoryId: String) -> Unit,
-    onScroll: (isTopBarVisible: Boolean) -> Unit
+    onScroll: (isTopBarVisible: Boolean) -> Unit,
+    categoriesScreenViewModel: CategoriesScreenViewModel = hiltViewModel()
+) {
+
+    val uiState by categoriesScreenViewModel.uiState.collectAsState()
+
+    when (val s = uiState) {
```

Comment:
I would like to do so, but Kotlin compiler emits error as it is not possible to smart cast uiSate. The code has [a composable function just to bind uiState to another name](https://github.com/android/nowinandroid/blob/0c542b42f956d363960f6d6e07ab35aa09064388/feature/interests/src/main/java/com/google/samples/apps/nowinandroid/feature/interests/InterestsScreen.kt#L45-L50) so that the author can use smart cast.  I am not sure having such function is a recommendation, and it seems slightly too much for me. What do you think?